### ISSUE TITLE
Does not build on musl libc

### ISSUE THREAD

**Issue by ddevault** [skyjake/lagrange#15]
This seems to be an issue with the foundation library but since there's no issue tracker for it I hope you won't mind me posting here.

```
[  1%] Building C object lib/the_Foundation/CMakeFiles/the_Foundation.dir/src/platform/posix/datagram.c.o
/home/sircmpwn/sources/lagrange/lib/the_Foundation/src/platform/posix/datagram.c: In function 'run_DatagramThread_':
/home/sircmpwn/sources/lagrange/lib/the_Foundation/src/platform/posix/datagram.c:108:9: error: unknown type name 'fd_set'
  108 |         fd_set reads, errors; {
      |         ^~~~~~
/home/sircmpwn/sources/lagrange/lib/the_Foundation/src/platform/posix/datagram.c:109:13: error: implicit declaration of function 'FD_ZERO'; did you mean 'NZERO'? [-Werror=implicit-function-declaration]
  109 |             FD_ZERO(&reads);
      |             ^~~~~~~
      |             NZERO
/home/sircmpwn/sources/lagrange/lib/the_Foundation/src/platform/posix/datagram.c:111:13: error: implicit declaration of function 'FD_SET'; did you mean 'L_SET'? [-Werror=implicit-function-declaration]
  111 |             FD_SET(output_Pipe(&d->wakeup), &reads);
      |             ^~~~~~
      |             L_SET
/home/sircmpwn/sources/lagrange/lib/the_Foundation/src/platform/posix/datagram.c:121:25: error: implicit declaration of function 'select' [-Werror=implicit-function-declaration]
  121 |             int ready = select(maxfd + 1, &reads, NULL, &errors, NULL);
      |                         ^~~~~~
/home/sircmpwn/sources/lagrange/lib/the_Foundation/src/platform/posix/datagram.c:127:13: error: implicit declaration of function 'FD_ISSET'; did you mean 'CPU_ISSET'? [-Werror=implicit-function-declaration]
  127 |         if (FD_ISSET(output_Pipe(&d->wakeup), &reads)) {
      |             ^~~~~~~~
      |             CPU_ISSET
```

**Comment by ddevault** [comment]
This is a trivial fix, just add `#include <sys/select.h>`

**Comment by skyjake** [comment]
Ok, I'll add that include.

I haven't announced or linked to this anywhere, but I've recently set up a mirror of [the_Foundation on Codeberg](https://codeberg.org/skyjake/the_Foundation) so issues can be submitted there.

**Comment by skyjake** [comment]
the_Foundation tests are having pthread-related crashes: https://codeberg.org/skyjake/the_Foundation/issues/1

**Comment by skyjake** [comment]
Fixed as of 26d797bca21ff157d87c4cc1f5fa59e17b1b8e16.
